### PR TITLE
predictor data preparation R script

### PR DESCRIPTION
**Pull request by krsnapaudel**
* First take on predictor data preparation. Need to handle other predictors. Mainly filter invalid values and apply certain transforms.
* slurm script to run on HPC.

For fpar and ndvi, we can make the script process one year at a time. For now, leaving it generic. For daily data, processing one year at a time (a stack of 365 rasters) may be too much.

Any R expertise would be helpful.

### CODE REVIEW

**Comment by krsnapaudel** [comment]
I realized making the repo private makes it difficult to work.

**Comment by ritviksahajpal** [comment]
My knowledge of R is very limited, but I can look at a written workflow description if it helps.

File: data_preparation/predictor_data_prep.r [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+library(terra)
+library(reshape2)
+library(stringr)
+
+valid_fpar_values <- function(fpar_rast) {
+  # check data_preparation/global_fpar_500m
+  # FPAR values: 0 to 100 (%). Valid range is 0-100.
+  # Flags: 251 is "other land", 254 is "water", 255 is "not processed".
+  fpar_rast[fpar_rast > 100] <- NA
+  fpar_rast[fpar_rast <= 0] <- NA
```
Comment (ellaampy) [inline data_preparation/predictor_data_prep.r:12]: @krsnapaudel fpar 's valid range includes 0

**Review by ellaampy — CHANGES_REQUESTED** [review]

File: data_preparation/predictor_data_prep.r (hunk 1 above)
Comment (krsnapaudel) [reply data_preparation/predictor_data_prep.r:12]: Yes, but zero does not add to much when multiplied or summed.

**Review by ellaampy — APPROVED** [review]

**Comment by krsnapaudel** [comment]
@ellaampy : The changes are not quite ready. Testing. I will upload new version soon.

**Comment by krsnapaudel** [comment]
The most recent update has two alternatives.
1. `predictor_data_prep.r`: Directly aggregates rasters.
2. `predictor_data_prep_v2.r`: Extracts to data.frame and aggregates.

We should use 1 as much as possible based on Dainius. Test 2 if 1 uses too much memory.

Tried to save memory by
1. Collapsing pretty much all function calls.
2. Overwriting variables with large amounts of data.

**Comment by krsnapaudel** [comment]
After testing in HPC. The first version is found to be faster. Removing version 2.

**Comment by krsnapaudel** [comment]
> My knowledge of R is very limited, but I can look at a written workflow description if it helps.

The workflow is as follows:

We work by group of countries for EU and FEWSNET.

Individual countries for the rest. This is because of shapefile sources.
```
for each indicator
  list files for a year
  create a raster stack of 50 files at a time
  replicate crop mask to have the same layers as rater stack
  process the raster stack (50 dates and all regions selected, could be multiple countries) at the same time
  multiply with crop mask and aggregate
  write to disk
```

**Pull request merged**